### PR TITLE
Backport: Support deletion of a large number of contracts

### PR DESCRIPTION
**Pull request by S11001001**
Backport of #10353 / 9c9b91eca7d5e8e9d391dec8000868ca4300f736. Part of #11641.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala [hunk 1]
```diff
@@ -394,6 +408,8 @@ private object PostgresQueries extends Queries {
 
   protected[this] override def jsonColumn(name: Fragment) = name ++ sql" JSONB NOT NULL"
 
+  protected[this] override val maxListSize = None
```
Comment (cocreature) [inline ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala:411]: could consider dropping the chunking given that it’s a noop here but I also get the value of not touching the backport more than needed :shrug:

**Review by cocreature — APPROVED** [review]
Thanks!

File: ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala [hunk 2]
```diff
@@ -214,15 +217,26 @@ sealed abstract class Queries {
...
+    import cats.instances.int._
+    import cats.syntax.foldable._
+    NonEmptyVector.fromVector(cids.toVector) match {
+      case None =>
+        free.connection.pure(0)
+      case Some(cids) =>
+        val chunks = maxListSize.fold(Vector(cids)) { size =>
+          require(size >= 1, s"size=$size but must be positive")
+          cids.toVector.grouped(size).map(NonEmptyVector.fromVectorUnsafe).toVector
+        }
```
Comment (S11001001) [inline ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala:234]: `NonEmptyVector#grouped` didn't exist at the time.

**Review by stefanobaghino-da — COMMENTED** [review]
Looks good, thanks.

**Pull request merged**